enclave/core/sgx/cpuid.c

```diff
@@ -78,19 +79,22 @@ int oe_emulate_cpuid(uint64_t* rax, uint64_t* rbx, uint64_t* rcx, uint64_t* rdx)
     // upper bits zeroed on 64-bit for CPUID
     uint32_t cpuid_leaf = (*rax) & 0xFFFFFFFF;
     uint32_t cpuid_sub_leaf = (*rcx) & 0xFFFFFFFF;
+    uint32_t index = 0;
 
-    if (cpuid_leaf < OE_CPUID_LEAF_COUNT &&
-        oe_is_emulated_cpuid_leaf(cpuid_leaf))
-    {
-        // For leaf 4 of cpuid, only subleaf of 0 is emulated
-        if ((cpuid_leaf == 4) && (cpuid_sub_leaf != 0))
-            return -1;
+    // emulate the cpuid leaves and find the corresponding index
+    while (index < OE_CPUID_LEAF_COUNT &&
```

Comment:
Similarly, why reimplement `oe_is_emulated_cpuid_leaf` here instead of just using the inline from the cpuid.h header?

Reply:
It is to get the index. `oe_is_emulated_cpuid_leaf` doesn't return the index. Or I can change the `oe_is_emulated_cpuid_leaf` interface to return the index.

Reply:
I would suggest defining `oe_get_emulated_cpuid_leaf` which returns the index on success (and -1 or `OE_UINT32_MAX` on failure) in internal/cpuid.h, and then redefining `oe_is_emulated_cpuid_leaf` as an inline/macro wrapper around `oe_get_emulated_cpuid_leaf` that converts the result to a `bool`.